### DISCUSSION TITLE
[Qt5] Project doesn't start when trying to instance QtQmlBot

### DISCUSSION THREAD

**Discussion by ruudschouten** [faaxm/spix#78] [General]
I've have installed AnyRPC and Spix, and have added the lines specified in the readme into my .pro file;

```qmake
...

# Spix
QT += quick
INCLUDEPATH += /usr/local/include
LIBS += -lSpix -lanyrpc

```
Building and running this by itself works fine, and doesn't produce any errors.
When I create a `spix::AnyRpcServer` instance with the default port, it also still works fine;
```cpp

int main(int argc, char *argv[])
{
    ...

    QQmlApplicationEngine *engine = new QQmlApplicationEngine;
    engine->addImportPath("qrc:/");

     // Set up Spix server
    spix::AnyRpcServer server;
    // auto bot = new spix::QtQmlBot();
    // bot->runTestServer(server);
    
    ...
}
```
but as soon as I uncomment the `auto bot = new spix::QtQmlBot();` line, I get the following errors in Qt Creator:

:-1: error: /usr/local/lib/libSpix.a(QtItemTools.cpp.o): in function `QByteArray::compare(char const*, Qt::CaseSensitivity) const':
QtItemTools.cpp:-1: error: undefined reference to `qstrnicmp(char const*, long long, char const*, long long)'
:-1: error: collect2: error: ld returned 1 exit status
:-1: error: [Makefile:82: mainapp] Error 1

And the following in the compile output
```log

/usr/bin/ld: /usr/local/lib/libSpix.a(QtItemTools.cpp.o): in function `QByteArray::compare(char const*, Qt::CaseSensitivity) const':
QtItemTools.cpp:(.text._ZNK10QByteArray7compareEPKcN2Qt15CaseSensitivityE[_ZNK10QByteArray7compareEPKcN2Qt15CaseSensitivityE]+0x63): undefined reference to `qstrnicmp(char const*, long long, char const*, long long)'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:82: mainapp] Error 1
make[2]: Leaving directory '/home/ruuds/Programming/djd/build-mea-Desktop_Qt_5_11_3_GCC_64bit-Debug/mainapp/mainapp'
make[1]: *** [Makefile:46: sub-mainapp-make_first] Error 2
make[1]: Leaving directory '/home/ruuds/Programming/djd/build-mea-Desktop_Qt_5_11_3_GCC_64bit-Debug/mainapp'
make: *** [Makefile:51: sub-mainapp-make_first] Error 2
15:59:58: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project mea (kit: Desktop Qt 5.11.3 GCC 64bit)
When executing step "Make"

```

Any help would be appreciated!

**Comment by 02deno** [comment]
I get the same error. Did you find a solution?

**Reply by ruudschouten** [reply]
Nope, I couldn't get it to work and decided to not use a UI testing solution for Qt
